vespa-http-client/src/main/java/com/yahoo/vespa/http/client/FeedClient.java

```diff
@@ -39,8 +41,21 @@
      * @param documentData the document data as JSON or XML (as specified when using the factory to create the API)
      * @param context a context object which will be accessible in the result of the callback, or null if none
      */
-    void stream(String documentId, CharSequence documentData, Object context);
+    default void stream(String documentId, CharSequence documentData, Object context) {
+        stream(documentId, null, documentData, context);
+    }
 
+    /**
+     * Streams a document to cluster(s). If the pipeline and buffers are full, this call will be blocking.
+     * Documents might time out before they are sent. Failed documents are not retried.
+     * Don't call stream() after close is called.
+     *
+     * @param documentId the document id of the document.
+     * @param operationId the id to use for this operation, or null to let the client decide an operation id
+     * @param documentData the document data as JSON or XML (as specified when using the factory to create the API)
+     * @param context a context object which will be accessible in the result of the callback, or null if none
+     */
+    void stream(String documentId, String operationId, CharSequence documentData, Object context);
```

Comment:
Yes, it would break mocking. I decided against doing something slightly uglier to avoid that since this goes into the second 7 release and I doubt anybody is mocking this in any case.